deepspeed/module_inject/replace_module.py

```diff
@@ -468,23 +429,24 @@ def replace_with_policy(child,
                     rotary_dim=rotary_dim,
                     mlp_after_attn=(rotary_dim is None or rotary_dim < 0),
                     mlp_act_func_type=policy.mlp_act_func_type,
-                    training_mp_size=training_mp_size,
+                    training_mp_size=config.training_mp_size,
                     bigscience_bloom=bigscience_bloom,
-                    max_out_tokens=max_out_tokens,
+                    max_out_tokens=config.max_out_tokens,
                     scale_attn_by_inverse_layer_idx=scale_attn_by_inverse_layer_idx)
 
-            if quantize and quantize_settings is not None:
+            if quantize and config.quantize_settings is not None:
                 (quantization_scales,
                  merge_count,
                  mlp_extra_grouping,
-                 quantize_groups) = quantize_settings
+                 quantize_groups) = config.quantize_settings
                 if moe:
                     new_module = transformer_inference.DeepSpeedMoEInference(
                         transformer_config,
-                        mp_group=mp_group,
-                        ep_group=None if ep_group is None else ep_group[num_experts],
-                        expert_mp_group=None
-                        if expert_mp_group is None else expert_mp_group[num_experts],
+                        mp_group=config.tensor_parallel.tp_group,
+                        ep_group=None if config.moe.ep_group is None else
+                        config.moe.ep_group[num_experts],
+                        expert_mp_group=None if config.moe.expert_mp_group is None else
```

Comment:
`config.moe.expert_mp_group` should be `config.moe.ep_mp_group`

Reply:
Actually, thinking a bit more about this new name. I think it should be called ```config.moe.expert_tp_group``` (as we use it for expert's tensor slicing or tensor parallelism degree). @RezaYazdaniAminabadi -- what do you think?

Reply:
We can decide on this later. For now, I just followed Mike's suggestion.